CodeSniffer/Standards/Generic/Sniffs/WhiteSpace/DisallowSpaceIndentSniff.php

```diff
@@ -94,42 +94,92 @@ public function process(PHP_CodeSniffer_File $phpcsFile, $stackPtr)
+            if ($tokens[$i]['code'] === T_INLINE_HTML && preg_match('`^(\s*)(\S.*)`s', $content, $matches) > 0) {
+                if (isset($matches[1]) === true) {
+                    $content = $matches[1];
+                }
+
+                if (isset($matches[2]) === true) {
+                    $nonWhitespace = $matches[2];
                 }
+            }
+
+            $hasSpaces = strpos($content, ' ');
+            $hasTabs   = strpos($content, "\t");
+
+            if ($hasSpaces === false && $hasTabs === false) {
+                // Empty line.
+                continue;
+            }
 
-                // Space are considered ok if they are proceeded by tabs and not followed
-                // by tabs, as is the case with standard docblock comments.
+            if ($hasSpaces === false && $hasTabs !== false) {
+                // All ok, nothing to do.
+                $phpcsFile->recordMetric($i, 'Line indent', 'tabs');
+                continue;
+            }
+
+            if ($tokens[$i]['code'] === T_DOC_COMMENT_WHITESPACE && $content === ' ') {
+                // Ignore file/class-level DocBlock, especially for recording metrics.
+                continue;
+            }
+
+            // OK, by now we know there will be spaces.
+            // We just don't know yet whether they need to be replaced or
+            // are precision indentation, nor whether they are correctly
+            // placed at the end of the whitespace.
+            $trimmed        = str_replace(' ', '', $content);
+            $numSpaces      = (strlen($content) - strlen($trimmed));
+            $numTabs        = (int) floor($numSpaces / $this->_tabWidth);
+            $tabAfterSpaces = strpos($content, "\t", $hasSpaces);
+
+            if ($hasTabs === false) {
                 $phpcsFile->recordMetric($i, 'Line indent', 'spaces');
-                $error = 'Tabs must be used to indent lines; spaces are not allowed';
-                $fix   = $phpcsFile->addFixableError($error, $i, 'SpacesUsed');
-                if ($fix === true) {
-                    $trimmed   = ltrim($content, ' ');
-                    $numSpaces = (strlen($content) - strlen($trimmed));
-                    if ($numSpaces < $this->_tabWidth) {
-                        $numTabs = 1;
-                        $padding = "\t";
-                    } else {
-                        $numTabs   = floor($numSpaces / $this->_tabWidth);
-                        $remaining = ($numSpaces - ($numTabs * $this->_tabWidth));
-                        $padding   = str_repeat("\t", $numTabs).$padding = str_repeat(' ', $remaining);
-                    }
 
-                    $phpcsFile->fixer->replaceToken($i, $padding.$trimmed);
+                if ($numTabs === 0) {
+                    // Ignore: precision indentation.
+                    continue;
+                }
+            } else {
+                if ($numTabs === 0) {
+                    // Precision indentation.
```

Comment:
This logic looks very suspicious to me (compare https://github.com/squizlabs/PHP_CodeSniffer/issues/547#issuecomment-92276735). Consider something like this:
```php
		if (! ($env === 'silent'
		       || ($env === 'production'
		       && ($level >= Minz_Log::NOTICE)))) {
```
Greatly increasing the tab width doesn't suffice as a workaround because then you'll get nonsensical errors about line length.